File: springboot-app/src/main/java/com/lumos/lumosspring/execution/entities/MaterialReservation.java
package com.lumos.lumosspring.execution.entities;

import com.lumos.lumosspring.stock.entities.MaterialStock;
import jakarta.persistence.*;

@Entity
@Table(name = "tb_material_reservation")
public class MaterialReservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_material_reservation")
    private long idMaterialReservation;

    @Column(columnDefinition = "TEXT")
    private String stockReservationName;

    @ManyToOne
    @JoinColumn(name = "material_stock_id")
    private MaterialStock materialStock;

    @ManyToOne
    private Street street;

    @Column(nullable = false)
    private int reservedQuantity;

    @Column(nullable = false, columnDefinition = "INTEGER DEFAULT 0")
    private int quantityCompleted;

    private ReservationStatus status; // Pode ser "pendente", "coletado", "cancelado"

    public long getIdMaterialReservation() {
        return idMaterialReservation;
    }

    public void setIdMaterialReservation(long idMaterialReservation) {
        this.idMaterialReservation = idMaterialReservation;
    }

    public String getStockReservationName() {
        return stockReservationName;
    }

    public void setStockReservationName(String stockReservationName) {
        this.stockReservationName = stockReservationName;
    }

    public MaterialStock getMaterialStock() {
        return materialStock;
    }

    public void setMaterialStock(MaterialStock material) {
        this.materialStock = material;
    }

    public int getReservedQuantity() {
        return reservedQuantity;
    }

    public void setReservedQuantity(int reservedQuantity) {
        this.reservedQuantity = reservedQuantity;
        this.materialStock.removeStockAvailable(reservedQuantity);
    }

    public int getQuantityCompleted() {
        return quantityCompleted;
    }

    public void setQuantityCompleted(int quantityCompleted) {
        this.quantityCompleted = quantityCompleted;
        this.materialStock.removeStockQuantity(quantityCompleted);
    }

    public ReservationStatus getStatus() {
        return status;
    }

    public void setStatus(ReservationStatus status) {
        this.status = status;
    }

    public enum ReservationStatus {
        PENDING,
        COLLECTED,
        CANCELLED
    }

}

File: springboot-app/src/main/java/com/lumos/lumosspring/execution/entities/MaterialReservation.kt
package com.lumos.lumosspring.execution.entities

import com.lumos.lumosspring.util.ReservationStatus
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table("material_reservation")
data class MaterialReservation(
    @Id
    var materialIdReservation: Long = 0,
    var description: String? = null,
    var centralMaterialStockId: Long? = null,
    var truckMaterialStockId: Long,
    var preMeasurementStreetId: Long? = null,
    var directExecution: Long? = null,
    var contractItemId: Long,
    var reservedQuantity: Double,
    var quantityCompleted: Double = 0.0,
    var status: String = ReservationStatus.PENDING,
    var teamId: Long,
    )

//fun confirmReservation() {
//    materialStock.removeStockAvailable(reservedQuantity)
//    status = ReservationStatus.APPROVED
//}
//
//fun rejectReservation() {
//    status = ReservationStatus.REJECTED
//}
//
//
//fun markAsCollected() {
//    status = ReservationStatus.COLLECTED
//    materialStock.removeStockQuantity(reservedQuantity)
//    team.deposit.materialStocks.find { it.material.idMaterial == materialStock.material.idMaterial }
//        ?.addStocks(reservedQuantity)
//}



